calculation/sha_calc/sha_calc/directivity/bea20/distributions.py

```diff
@@ -0,0 +1,314 @@
+    dist_range = upper - lower
+    truncated_points = (np.random.uniform(0, 1, hypo_along_strike)) * dist_range + lower
+    truncated_distribution = strike_distribution.ppf(truncated_points)
+    distances = truncated_distribution * total_length
+
+    # Works out the depth method for down dip placement of hypocentres
+    # Based on Weilbull or Gamma distributions depending on the EventType
+    if event_type == EventType.DIP_SLIP:
+        distribution = stats.gamma(a=7.364, scale=0.072)
+    elif event_type == EventType.STRIKE_SLIP:
+        distribution = stats.weibull_min(scale=0.626, c=3.921)
+    else:
+        distribution = stats.weibull_min(scale=0.612, c=3.353)
+
+    # Truncate between 0 and 1 for hypocentre depth to ensure none exceed the boundaries
+    upper, lower = distribution.cdf((1, 0))
+    dist_range = upper - lower
+    truncated_points = (np.random.uniform(0, 1, hypo_down_dip)) * dist_range + lower
+    down_dip_distribution = distribution.ppf(truncated_points)
+
+    planes_list = []
+    planes_index = []
+
+    for distance in distances:
+        current_length = 0
+        planes_copy = [plane.copy() for plane in planes]
+        for index, plane in enumerate(planes_copy):
+            if current_length < distance < (current_length + plane["length"]):
+                for depth in down_dip_distribution:
+                    planes_depth_copy = [plane.copy() for plane in planes]
+                    planes_depth_copy[index]["shyp"] = distance - (total_length / 2)
+                    planes_depth_copy[index]["dhyp"] = plane["width"] * depth
+                    planes_index.append(index)
+                    planes_list.append(planes_depth_copy)
+            current_length += plane["length"]
+
+    return planes_list, planes_index, []
+
+
+def even_grid(
+    hypo_along_strike: int,
+    hypo_down_dip: int,
+    planes: List,
+    event_type: EventType,
+    total_length: float,
+):
+    """
+    Uses an evenly spaced grid across strike and dip based on the amount of hypocentres specified across both lengths
+
+    Parameters
+    ----------
+    hypo_along_strike: int
+        Number of hypocentres across strike
+    hypo_down_dip: int
+        Number of hypocentres down dip
+    planes: list
+        The planes to adjust and set the hypocentre on
+    event_type: EventType
+        The event type Strike_slip, dip_slip or all for determining the down dip distribution function
+    total_length: float
+        The total length of the fault
+    """
+
+    gap = total_length / (hypo_along_strike + 1)
+    distances = [gap * (x + 1) for x in range(hypo_along_strike)]
+
+    down_gap = planes[0]["width"] / (hypo_down_dip + 1)
+    down_dip_values = [down_gap * (x + 1) for x in range(hypo_down_dip)]
+
+    # Works out the distances across strike of the fault for each hypocentre
+    # Based on a normal distribution and truncated between 0 and 1
+    mu, sigma = 0.5, 0.23
```

Comment:
I think it is slightly easier to read if these are assigned on separate lines. @claudio525 might say otherwise as they are closely correlated.

Reply:
Haha yeah I think thats perfectly fine, but yeah kinda personal preference